File: Contents/SevenSwitch_VisualLizer/SevenSwitch_iOS/Examples/ViewControllerA.m
//
//  ViewControllerA.m
//  MGUSevenSwitch
//
//  Created by Kwan Hyun Son on 27/01/2020.
//  Copyright © 2020 Mulgrim Inc. All rights reserved.
//

#import "ViewControllerA.h"
@import IosKit;

@interface ViewControllerA ()
@property (nonatomic, strong) MGUSevenSwitch *sevenSwitch;
@property (nonatomic, strong) id <NSObject>sevenSwitchObserver;
@end

@implementation ViewControllerA

- (void)dealloc {
    NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
    [nc removeObserver:_sevenSwitchObserver];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = @"MGUSevenSwitch를 만들었다.";
    
    CGPoint center = CGPointMake(self.view.bounds.size.width * 0.5, self.view.bounds.size.height * 0.5);
    
    _sevenSwitch = [[MGUSevenSwitch alloc] initWithCenter:center
                                                 switchOn:YES
                                            configuration:[MGUSevenSwitchConfiguration defaultConfiguration]];
    [self.sevenSwitch addTarget:self action:@selector(switchChanged:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:self.sevenSwitch];
    [self.sevenSwitch mgrPinCenterToSuperviewCenterWithFixSize:self.sevenSwitch.frame.size];
    
    __weak __typeof(self) weakSelf = self;
    NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
    _sevenSwitchObserver = [nc addObserverForName:MGUSevenSwitchStateChangedNotification
                                           object:self.sevenSwitch // poster
                                            queue:[NSOperationQueue mainQueue]
                                       usingBlock:^(NSNotification *note) {
        // NSNumber *duration = note.userInfo[UIKeyboardAnimationDurationUserInfoKey];
        NSLog(@"바뀌었냐?? %d", weakSelf.sevenSwitch.switchOn);
    }];
    
    UILabel *label = [UILabel new];
    label.text = @"MGUSevenSwitch";
    label.font = [UIFont systemFontOfSize:10.0];
    [self.view addSubview:label];
    label.translatesAutoresizingMaskIntoConstraints = NO;
    [label.trailingAnchor constraintEqualToAnchor:self.sevenSwitch.trailingAnchor].active = YES;
    [label.topAnchor constraintEqualToAnchor:self.sevenSwitch.bottomAnchor constant:55.0].active = YES;
}

- (IBAction)sevenSwitch1ValueChanged:(MGUSevenSwitch *)sender {
    if (sender.switchOn == YES) {
        NSLog(@"밸류가 바뀌었네. ON");
    } else {
        NSLog(@"밸류가 바뀌었네. OFF");
    }
}

- (void)switchChanged:(MGUSevenSwitch *)sender {
    if (sender.switchOn == YES) {
        NSLog(@"밸류가 바뀌었네. ON");
    } else {
        NSLog(@"밸류가 바뀌었네. OFF");
    }
}


#pragma mark - 버튼 액션 메서드

- (IBAction)onWithAnimated:(id)sender {
    [self.sevenSwitch setSwitchOn:YES withAnimated:YES];
}
- (IBAction)offWithAnimated:(id)sender{
    [self.sevenSwitch setSwitchOn:NO withAnimated:YES];
}
- (IBAction)onWithNoAnimated:(id)sender{
    [self.sevenSwitch setSwitchOn:YES withAnimated:NO];
}
- (IBAction)offWithNoAnimated:(id)sender{
    [self.sevenSwitch setSwitchOn:NO withAnimated:NO];
}

@end

File: Contents/MGUFlatSwitch/MGUFlatSwitch/Examples/ViewControllerA.swift
//
//  ViewControllerA.swift
//  MGUFlatSwitch
//
//  Created by Kwan Hyun Son on 2023/05/10.
//

import UIKit
import IosKit

final class ViewControllerA: UIViewController {
    
    // MARK: - Property
    @IBOutlet private weak var flatSwitch: MGUFlatSwitch!
    private var flatSwitch2 = MGUFlatSwitch(frame: CGRectMake(0.0, 0.0, 100.0, 100.0))
    
    override func viewDidLoad() {
        super.viewDidLoad()
        title = "MGUFlatSwitch Samples"
        flatSwitch.addTarget(self, action:#selector(switchClicked(_:)), for: .valueChanged)
        
        flatSwitch2.lineWidth = 9.0
        
        let baseCircleStrokeColor =
        UIColor.mgrColor(withLightModeColor: UIColor(red: 0.3, green: 0.3, blue: 0.6, alpha: 0.5),
                         darkModeColor: UIColor.red.withAlphaComponent(0.5),
                         darkElevatedModeColor: nil)
        flatSwitch2.baseCircleStrokeColor = baseCircleStrokeColor
        
        let checkMarkNCircleStrokeColor =
        UIColor.mgrColor(withLightModeColor: UIColor(red: 0.3, green: 0.3, blue: 0.6, alpha: 1.0),
                         darkModeColor: .red,
                         darkElevatedModeColor: nil)
        flatSwitch2.checkMarkNCircleStrokeColor = checkMarkNCircleStrokeColor
        view.addSubview(flatSwitch2)
        flatSwitch2.mgrPinCenterToSuperviewCenter(withFix: CGSizeMake(150.0, 150.0))
        
        let flatSwitch3 = MGUFlatSwitch(frame: CGRectMake(0.0, 0.0, 50.0, 50.0))
        flatSwitch3.lineWidth = 3.0
        flatSwitch3.baseCircleStrokeColor = baseCircleStrokeColor
        flatSwitch3.checkMarkNCircleStrokeColor = checkMarkNCircleStrokeColor
        view.addSubview(flatSwitch3)
        flatSwitch3.mgrPinFix(CGSize(width: 50.0, height: 50.0))
        flatSwitch3.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
        flatSwitch3.topAnchor.constraint(equalTo: flatSwitch2.bottomAnchor, constant: 50.0).isActive = true
        
    }
    
    @objc private func switchClicked(_ sender: MGUFlatSwitch) {
        print("터치", "\(sender.isSelected)")
    }

}
